### PR TITLE
Do not override StorageChecker.errors in StorageSpoke (#1252596)

### PR DESCRIPTION
**Pull request by vojtechtrefny**
When assigning checkDiskSelection result to self.errors, we can
lose errors from CustomPartitioningSpoke. Added separate "category"
for disk selection errors to fix this.

### CODE REVIEW

**Comment by vpodzime** [comment]
Looks good to me.

**Comment by AdamWill** [comment]
Uh. What about all the other places where the file checks for `self.errors` - i.e. `completed()`, `status()`, `refresh()`? This isn't going to do the right thing there, is it? The `disk_errors` will be disregarded.

**Comment by vojtechtrefny** [comment]
> Uh. What about all the other places where the file checks for self.errors - i.e. completed(), status(), refresh()? This isn't going to do the right thing there, is it? The disk_errors will be disregarded.

These errors are specific just for the case when your disk selection is "wrong". The `checkDiskSelection` method returns errors only if you select just subset of disks with raid/lvm/btrfs. In this case you need either to select them all or quit the installation. It is not possible to leave the `StorageSpoke` with this selection and the check is performed every time you try to leave it (using the "Done" button).

File: pyanaconda/ui/gui/spokes/storage.py [hunk 1]
```diff
@@ -1014,7 +1015,28 @@ def on_specialized_clicked(self, button):
...
+                errors = "\n".join(self.disks_errors)
+                dialog.refresh(errors)
+                rc = dialog.run()
+
+            dialog.window.destroy()
+
+            if rc == 0:
+                # Quit.
+                iutil.ipmi_report(constants.IPMI_ABORTED)
+                sys.exit(0)
```
Comment (jkonecny12) [inline pyanaconda/ui/gui/spokes/storage.py]: Could you please move the same code (below) to separate method and call it instead.

Comment (vojtechtrefny) [inline pyanaconda/ui/gui/spokes/storage.py]: Actually the same code is not as long -- the `dialog` is different (different buttons), `label` is different and `errors` are different. I'm not sure that a new method just for this case with five parameters would be better/more clear.

Comment (jkonecny12) [inline pyanaconda/ui/gui/spokes/storage.py]: Yeah right, sorry I missed this differences before. 
Looks good to me (now when I'm not so blind :) ), I'm returning the Ack now.

**Comment by vojtechtrefny** [comment]
Comment added to explain why errors from `checkDiskSelection` can be in separate list and not in `self.errors`.

**Comment by vpodzime** [comment]
Looks even better to me now, thanks!

**Comment by vojtechtrefny** [comment]
Pushed.

Still not convinced if a special method is needed here, but I can always do a follow-up fixing this.

**Pull request closed without merging**